x-pack/plugins/ml/public/application/components/import_export_jobs/import_jobs_flyout/import_jobs_flyout.tsx

```diff
@@ -0,0 +1,493 @@
+    setDfaJobs([]);
+    setJobIds([]);
+    setIdsMash('');
+    setImporting(false);
+    setJobType(null);
+    setTotalJobsRead(0);
+    setValidatingJobs(false);
+    setShowFileReadError(showFileError);
+  }, []);
+
+  useEffect(() => {
+    reset();
+  }, [showFlyout]);
+
+  function toggleFlyout() {
+    setShowFlyout(!showFlyout);
+  }
+
+  const onFilePickerChange = useCallback(async (files: any) => {
+    setShowFileReadError(false);
+
+    if (files.length === 0) {
+      reset();
+      return;
+    }
+
+    try {
+      const loadedFile = await readJobConfigs(files[0]);
+      if (loadedFile.jobType === null) {
+        reset(true);
+        return;
+      }
+
+      setTotalJobsRead(loadedFile.jobs.length);
+
+      const validatedJobs = await validateJobs(
+        loadedFile.jobs as ImportedAdJob[],
+        loadedFile.jobType,
+        getIndexPatternTitles
+      );
+
+      if (loadedFile.jobType === 'anomaly-detector') {
+        const tempJobs = (loadedFile.jobs as ImportedAdJob[]).filter((j) =>
+          validatedJobs.jobIds.includes(j.job.job_id)
+        );
+        setAdJobs(tempJobs);
+      } else if (loadedFile.jobType === 'data-frame-analytics') {
+        const tempJobs = (loadedFile.jobs as DataFrameAnalyticsConfig[]).filter((j) =>
+          validatedJobs.jobIds.includes(j.id)
+        );
+        setDfaJobs(tempJobs);
+      }
+
+      setJobType(loadedFile.jobType);
+      setJobIds(
+        validatedJobs.jobIds.map((id) => ({
+          id,
+          originalId: id,
+          valid: true,
+          invalidMessage: '',
+        }))
+      );
+      setIdsMash(validatedJobs.jobIds.map((id) => id).join(''));
+      setValidatingJobs(true);
+      setSkippedJobs(validatedJobs.skippedJobs);
+    } catch (error) {
+      displayErrorToast(error);
+    }
+  }, []);
+
+  const onImport = useCallback(async () => {
+    setImporting(true);
+    if (jobType === 'anomaly-detector') {
+      const renamedJobs = renameAdJobs(jobIds, adJobs);
+      try {
+        await bulkCreateADJobs(renamedJobs);
+      } catch (error) {
+        // display unexpected error
+        displayErrorToast(error);
+      }
+    } else if (jobType === 'data-frame-analytics') {
+      const renamedJobs = renameDfaJobs(jobIds, dfaJobs);
+      await bulkCreateDfaJobs(renamedJobs);
+    }
+
+    setImporting(false);
+    setShowFlyout(false);
+    refreshJobs();
+  }, [jobType, jobIds, adJobs, dfaJobs]);
+
+  const bulkCreateADJobs = useCallback(async (jobs: ImportedAdJob[]) => {
+    const results = await bulkCreateJobs(jobs);
+    let successCount = 0;
+    Object.entries(results).forEach(([jobId, { job, datafeed }]) => {
+      if (job.error || datafeed.error) {
+        if (job.error) {
+          const title = i18n.translate('xpack.ml.importExport.importFlyout.importADJobError', {
+            defaultMessage: 'Could not create job {jobId}',
+            values: { jobId },
+          });
+          displayErrorToast(job.error, title);
```

Comment:
yeah, I thought it would be manageable to have the user deal with each toast one at a time, but this won't be nice if they're attempting to import hundreds of jobs.
I'll collect up all of the errors and add them to a single toast.

Reply:
Fixed in https://github.com/elastic/kibana/pull/101037/commits/f45f3cfcabae2d31cf43ae50895795b7fe9e5e74
All errors are collected and displayed in a single toast.